src/cargo/core/compiler/mod.rs

```diff
@@ -211,6 +211,7 @@ fn rustc<'a, 'cfg>(
     // If we are a binary and the package also contains a library, then we
     // don't pass the `-l` flags.
     let pass_l_flag = unit.target.is_lib() || !unit.pkg.targets().iter().any(|t| t.is_lib());
+    let pass_link_args = unit.target.is_cdylib();
```

Comment:
Could this perhaps be renamed to `pass_cdylib_link_args`?

Reply:
I can rename it and fold it in the last patch or make another, I'm a little wary of last minute changes :)